src/System.CommandLine.Hosting.Tests/HostingTests.cs

```diff
@@ -191,5 +191,39 @@ void Execute(IHost host)
 
             testConfigValue.Should().BeEquivalentTo(testValue);
         }
+
+        [Fact]
+        public static void UseHost_binds_parsed_arguments_to_options()
+        {
+            const int myValue = 4224;
+            string commandLine = $"-{nameof(MyOptions.MyArgument)} {myValue}";
+
+            var rootCmd = new RootCommand();
+            rootCmd.AddOption(
+                new Option($"-{nameof(MyOptions.MyArgument)}")
+                { Argument = new Argument<int>() }
+                );
+            rootCmd.Handler = CommandHandler.Create((IHost host) =>
+            {
+                var options = host.Services.GetRequiredService<IOptions<MyOptions>>();
+                Assert.Equal(myValue, options.Value.MyArgument);
```

Comment:
Generally it's a bit easier to read if you just capture the value here and assert at the end of the test. This also avoids the problem of the handler, and therefore the assertion, not being called.

Reply:
@jonsequitur https://github.com/dotnet/command-line-api/pull/664/commits/aca8727feede0e8ec29ae2e2dff84f76522e46d8